### ISSUE TITLE
Syncthing floods standard output with 'invalid URL escape' messages

### ISSUE THREAD

**Issue by ms2048** [syncthing/syncthing#7581]
Similar bugs: #2709 #5827 are closed but very similar error messages are repeatedly generated.

Windows 7 Ultimate (with changed locale from Polish to English - maybe this is important)
Syncthing v1.15.1

The messages are all similar:
`[...] INFO: Parsing dialer address tcp://[fe80::424e:36ff:fe1a:9bd2%25Po%C5%82%C4%85czenie%20lokalne]:22000: parse "tcp://[fe80::424e:36ff:fe1a:9bd2%25Po%C5%82%C4%85czenie%20lokalne]:22000": invalid URL escape "%C5"`

'Połączenie lokalne' means 'Local connection' in Polish and is a default name for a wired connection on Windows.

I'm new to Syncthing (using it for about a month on a few devices) and maybe I am doing something wrong, in this case I apologize.

**Comment by bt90** [comment]
https://tools.ietf.org/html/rfc4007#section-11.7

> Also, the format for non-global addresses might conflict with the URI syntax [13], since the syntax defines the delimiter character (`%') as the escape character.  This conflict would require, for example, that the <zone_id> part for zone 1 with the delimiter be represented as '%251'.  It also means that we could not simply copy a non-escaped format from other sources as input to the URI parser.  Additionally, if the URI parser does not convert the escaped format before passing it to a name-to-address library, the conversion will fail.  All these issues would decrease the benefit of the textual representation described in this section.

**Comment by calmh** [comment]
Yes, we handle the required escaping properly, the issue here seems to be that even escaped non-ASCII characters aren't accepted.

**Comment by bt90** [comment]
@calmh is the URI being parsed here one published by some other device?

**Comment by calmh** [comment]
It's the source address of the incoming packet as reported by `(*net.UDPConn).ReadFrom()` (or such) put into the host portion of a `net.URL`.

**Comment by bt90** [comment]
Is the link local IP dialable without the identifier?

**Comment by calmh** [comment]
Certainly not. (It's link scoped.)

**Comment by bt90** [comment]
Does golang provide something comparable to `if_nametoindex` ? Looks like IPv6 link local addresses can also be expressed with the index of the network interface:

fe80::4d3d:3426:46e6:1457%2

**Comment by calmh** [comment]
You could check it out, this seems to be an issue close to your interests. :)

**Comment by AudriusButkevicius** [comment]
I'm not conviced that is true on windows.

**Comment by bt90** [comment]
> I'm not conviced that is true on windows.

`ipconfig` lists my link local IPv6 address by default with the index 😉 

![grafik](https://user-images.githubusercontent.com/1926175/115062476-aacc8c80-9eea-11eb-9f77-b5389d2e1818.png)

**Comment by bt90** [comment]
Seems like this mapping is also supported by the go standard library: https://golang.org/pkg/net/#Interface
